samples/ordered_puzzle/components/ordered_puzzle.js

```diff
@@ -0,0 +1,94 @@
+AFRAME.registerComponent("ordered-puzzle", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+  },
+
+  multiple: true,
+
+  init: function () {
+    const data = this.data.jsonData;
+    const el = this.el;
+
+    const numPuzzlePieces = data.images.length;
+
+    for (i = 0; i < numPuzzlePieces; i++) {
+      var textBoxProp =
+        '{"width": "' +
+        data.images[i].width +
+        '", "height": "' +
+        data.images[i].height +
+        '", "color": "yellow", "x": "' +
+        data.images[i].xTarget +
+        '", "y": "' +
+        data.images[i].yTarget +
+        '", "z": "0", "text": "' +
+        i +
+        '"}';
+
+      this.target = document.createElement("a-entity");
+      this.target.setAttribute("id", "puzzle-target");
```

Comment:
This will cause duplicate id for all the puzzle pieces, so let's add an extra identifier, like "puzzle-target-" + i